packages/edit-site/src/components/welcome-guide/template.js

```diff
@@ -10,29 +10,23 @@ import { store as editorStore } from '@wordpress/editor';
 /**
  * Internal dependencies
  */
-import { store as editSiteStore } from '../../store';
+import useEditedEntityRecord from '../use-edited-entity-record';
 
 export default function WelcomeGuideTemplate() {
 	const { toggle } = useDispatch( preferencesStore );
 
-	const isVisible = useSelect( ( select ) => {
-		const isTemplateActive = !! select( preferencesStore ).get(
-			'core/edit-site',
-			'welcomeGuideTemplate'
-		);
-		const isEditorActive = !! select( preferencesStore ).get(
-			'core/edit-site',
-			'welcomeGuide'
-		);
-		const { isPage } = select( editSiteStore );
-		const { getCurrentPostType } = select( editorStore );
-		return (
-			isTemplateActive &&
-			! isEditorActive &&
-			isPage() &&
-			getCurrentPostType() === 'wp_template'
-		);
+	const edited = useEditedEntityRecord();
+	const isPostTypeTemplate = edited.record.type === 'wp_template';
+	const { isActive, hasPreviousEntity } = useSelect( ( select ) => {
+		const { getEditorSettings } = select( editorStore );
+		const { get } = select( preferencesStore );
+		return {
+			isActive: get( 'core/edit-site', 'welcomeGuideTemplate' ),
+			hasPreviousEntity:
+				!! getEditorSettings().onNavigateToPreviousEntityRecord,
+		};
 	}, [] );
```

Comment:
Completely optional, just a bit neater:

```suggestion
	const { isActive, hasPreviousEntity } = useSelect(
		( select ) => ( {
			isActive: select( preferencesStore ).get(
				'core/edit-site',
				'welcomeGuideTemplate'
			),
			hasPreviousEntity:
				!! select( editorStore ).getEditorSettings()
					.onNavigateToPreviousEntityRecord,
		} ),
		[]
	);
```

Reply:
I waffle on this. On one hand, I do appreciate assignments aren’t needed. On the other, in many cases I would prefer to leave them there in hopes it helps avoid any future additions of `select` calls that reproduce one of the existing ones—not that it guarantees it or a duplicate is too concerning.

Therefore, fewer lines wins out for me 😝.

Reply:
Yeah I get you. Either way is fine in the end as it's all transpiled into goo 😄